packages/block-library/src/button/edit.native.js

```diff
@@ -172,22 +172,20 @@ class ButtonEdit extends Component {
 	}
 
 	getBackgroundColor() {
-		const { backgroundColor, attributes } = this.props;
-		const { style } = attributes;
+		const { backgroundColor, wrapperProps } = this.props;
 
 		return (
-			( style && style.color && style.color.background ) ||
+			wrapperProps?.style?.backgroundColor ||
 			backgroundColor.color ||
```

Comment:
Do we still need `backgroundColor.color` ? Is it for backward compatibility ? If so, it could be good to drop an inline comment to explain why we still need that.

Reply:
Yes, we need that for supporting colors that are not custom ones and are set from the color palette. Let me check if i could merge them in the `block.native.js` and we could remove it from here. Additional pros of this is that will be applied for all blocks (like media-text etc)